storm-core/src/jvm/org/apache/storm/metric/ClusterMetricsConsumerExecutor.java

```diff
@@ -40,31 +39,49 @@ public ClusterMetricsConsumerExecutor(String consumerClassName, Object registrat
     public void prepare() {
         try {
             metricsConsumer = (IClusterMetricsConsumer)Class.forName(consumerClassName).newInstance();
+            metricsConsumer.prepare(registrationArgument);
         } catch (Exception e) {
-            throw new RuntimeException("Could not instantiate a class listed in config under section " +
-                    Config.STORM_CLUSTER_METRICS_CONSUMER_REGISTER + " with fully qualified name " + consumerClassName, e);
-        }
+            LOG.error("Could not instantiate or prepare Cluster Metrics Consumer with fully qualified name " +
+                    consumerClassName, e);
 
-        metricsConsumer.prepare(registrationArgument);
+            if (metricsConsumer != null) {
+                metricsConsumer.cleanup();
+            }
+            metricsConsumer = null;
+        }
     }
 
     public void handleDataPoints(final IClusterMetricsConsumer.ClusterInfo clusterInfo, final Collection<DataPoint> dataPoints) {
-        try {
-            metricsConsumer.handleDataPoints(clusterInfo, dataPoints);
-        } catch (Throwable e) {
-            LOG.error("Error while handling cluster data points, consumer class: " + consumerClassName, e);
+        if (metricsConsumer == null) {
```

Comment:
Some modules (at least what I've reported) could throw error on initialization phase when backend storage is not working properly. This could be intermittent so I thought it might be better to retry when needed, but I'm also OK to remove retry logic and leave error log if we don't want to retry automatically.